src/web3id.rs

```diff
@@ -0,0 +1,797 @@
+                Bytes(
+                    hex::decode("575851a4e0558d589a57544a4a9f5ad1bd8467126c1b6767d32f633ea03380e6")
+                        .unwrap(),
+                ),
+            ),
+            (
+                AttributeTag::FirstName,
+                Bytes(
+                    hex::decode("575851a4e0558d589a57544a4a9f5ad1bd8467126c1b6767d32f633ea03380e6")
+                        .unwrap(),
+                ),
+            ),
+        ];
+        let commitment_inputs = VerifiableCredentialCommitmentInputs::Account {
+            issuer: 1,
+            values: values.into_iter().collect(),
+            randomness: randomness.into_iter().collect(),
+        };
+        web3id::OwnedCommitmentInputs::try_from(commitment_inputs.clone())
+            .expect("Can convert commitment inputs to base type");
+
+        create_verifiable_presentation(request, get_global(), vec![commitment_inputs])
+            .expect("Can create presentation");
+    }
+
+    #[test]
+    fn test_create_verifiable_presentation_web3id() {
+        let request = VerifiablePresentationRequest {
+            challenge: Bytes(
+                hex::decode("94d3e85bbc8ff0091e562ad8ef6c30d57f29b19f17c98ce155df2a30100dAAAA")
+                    .unwrap(),
+            ),
+            statements: vec![VerifiableCredentialStatement::Web3Id {
+                cred_type: vec![],
+                network: Network::Testnet,
+                contract: ContractAddress {
+                    index: 1,
+                    subindex: 0,
+                },
+                holder_id: Bytes(
+                    hex::decode("3ea04bafe5227cb26e4e9ff388e6d83ece3f7de558a242e2a84015f4fa36f859")
+                        .unwrap(),
+                ),
+                statement: vec![
+                    AtomicStatementV2::AttributeInSet {
+                        statement: AttributeInSetStatementV2 {
+                            attribute_tag: "degreeType".to_string(),
+                            set: vec![
+                                Web3IdAttribute::String {
+                                    value: "BachelorDegree".to_string(),
+                                },
+                                Web3IdAttribute::String {
+                                    value: "MasterDegree".to_string(),
+                                },
+                            ],
+                        },
+                    },
+                    AtomicStatementV2::RevealAttribute {
+                        statement: RevealAttributeStatementV2 {
+                            attribute_tag: "degreeName".to_string(),
+                        },
+                    },
+                ],
+            }],
+        };
+
+        web3id::Request::try_from(request.clone()).expect("Can convert request to base type");
+
+        let values: Vec<(String, Web3IdAttribute)> = vec![
+            (
+                "degreeName".to_string(),
+                Web3IdAttribute::String {
```

Comment:
I'll add tests for converting between the input/output types and the corresponding `concordium_base` types, but IMO other tests will be mostly redundant as this will already have tests in the modules in base.

Reply:
I added tests that ensure the conversion to/from `web3id::Web3IdAttribute` works for all variants of `Web3IdAttribute`. This should be sufficient as tests for generating presentations for different variants should be covered in `concordium_base`

Reply:
I added the remaining unrepresented attributes to the presentation creation tests as well 😄